### PR TITLE
fix: properly parsing numbers that are strings

### PR DESCRIPTION
**Pull request by Datron**
## Problem
In cases where a user enters as input a string which is a number, our parsing fails

## Solution
add support for it

### CODE REVIEW

File: crates/frontend/src/utils.rs [hunk 1]
```diff
@@ -313,6 +313,7 @@ pub fn parse_value(val: &Value, config_type: ConfigValueType) -> Result<Value, S
 
         ConfigValueType::String => match val {
             Value::String(_) => Ok(val.clone()),
+            Value::Number(i) => Ok(Value::String(i.to_string())),
```
Comment (ayushjain17) [inline crates/frontend/src/utils.rs:316]: can we also add for bool as well

File: crates/frontend/src/components/context_form/utils.rs [hunk 2]
```diff
@@ -186,7 +187,15 @@ pub fn construct_context(
     } else {
         let condition_schemas = conditions
             .iter()
-            .map(|condition| get_condition_schema(condition, dimensions.clone()).unwrap())
+            .map(
```
Comment (ayushjain17) [inline crates/frontend/src/components/context_form/utils.rs:190]: shouldn't this be a filter_map ?
for `Ok(ans)` we can return `Some(ans)`
and for `Err` we can return `None` directly, returning `Null` [here](https://github.com/juspay/superposition/pull/285/files#diff-564227801c44858d95faa622d6194be18193e660848f20a138956b1fd2a15efbR195) seemed wrong

**Review by ayushjain17 — CHANGES_REQUESTED** [review]

File: crates/frontend/src/utils.rs (hunk 1 above)
Comment (Datron) [reply crates/frontend/src/utils.rs:316]: Sure makes sense

File: crates/frontend/src/components/context_form/utils.rs (hunk 2 above)
Comment (Datron) [reply crates/frontend/src/components/context_form/utils.rs:190]: I want the parsing to fail with Null, I don't want to filter out failed values and still proceed. I want to fail on wrong inputs

**Review by mahatoankitkumar — APPROVED** [review]

**Review by ayushjain17 — APPROVED** [review]

**Pull request merged**